streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -361,7 +351,13 @@ public boolean process() {
             log.trace("Start processing one record [{}]", record);
 
             updateProcessorContext(record, currNode);
-            currNode.process(record.key(), record.value());
+
+            StreamsMetricsImpl.maybeMeasureLatency(
```

Comment:
I've did some investigation and now I think using this lambda-function-pattern is fine with negligible cost.